### PR TITLE
Update the repo, so it builds with modern Bazel and can be worked on

### PR DESCRIPTION
**Pull request by jjudd**
* Update Bazel to 7.1.1 and use Bazelisk to launch it
* Update 3rd party Bazel dependencies
* Move to GitHub Actions from Travis
* Update to Scala 3
* Update Scala 3rd party dependencies
* Update Maven publishing to be like our Play Routes Compiler CLI
* Add Bazel formatting and linting with buildifier

### CODE REVIEW

**Comment by jjudd** [comment]
This is a rather broad PR, so I wouldn't be surprised if there's something off.

I also imagine that I won't get the Travis -> GitHub Actions move right on the first try, so there'll probably be some more commits flowing in to this PR while I get that working.

**Comment by jjudd** [comment]
@coreywoodfield you don't show up in the reviewers drop down, but your review would be appreciated.

File: scripts/gen-deps.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+#
+# Regenerates the external dependencies lock file using rules_jvm_external
+#
+
+cd "$(dirname "$0")/.."
+echo "$(dirname "$0")/.."
+
+echo "generating dependencies for main workspace"
+bazel run @unpinned_twirl_compiler_cli_maven//:pin
+ bazel run @unpinned_twirl_compiler_cli_test_maven//:pin
```
Comment (coreywoodfield) [inline scripts/gen-deps.sh]:
```suggestion
bazel run @unpinned_twirl_compiler_cli_test_maven//:pin
```

File: scripts/publish.sh [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#! /bin/bash
+#! /bin/bash -e
 ##
-## Build and deploy the twirl compiler jar
+## Build and deploy the play routes compiler jar
```
Comment (coreywoodfield) [inline scripts/publish.sh]: Why change this?

**Review by coreywoodfield — APPROVED** [review]

File: scripts/gen-deps.sh (hunk 1 above)
Comment (jjudd) [reply scripts/gen-deps.sh]: Good catch. Fixed.

File: scripts/publish.sh (hunk 2 above)
Comment (jjudd) [reply scripts/publish.sh]: Copy pasta mistake

Comment (jjudd) [reply scripts/publish.sh]: Fixed

**Pull request merged**